login_server/src/cpp/JSONInterface/JsonUpdateUserInfos.cpp

```diff
@@ -144,32 +144,75 @@ Poco::JSON::Object* JsonUpdateUserInfos::handle(Poco::Dynamic::Var params)
 					}
 				}
 			}
-			else if ("User.password" == name && value.size() > 0 && (ServerConfig::g_AllowUnsecureFlags & ServerConfig::UNSECURE_PASSWORD_REQUESTS) == ServerConfig::UNSECURE_PASSWORD_REQUESTS) {
+			else if ("User.password" == name && (ServerConfig::g_AllowUnsecureFlags & ServerConfig::UNSECURE_PASSWORD_REQUESTS) == ServerConfig::UNSECURE_PASSWORD_REQUESTS) {
+				
 				if (!value.isString()) {
 					jsonErrorsArray.add("User.password isn't string");
 				}
+				std::string value_str = value.toString();
+				if (!value_str.size()) {
+					jsonErrorsArray.add("User.password is empty");
+				}
 				else {
-					NotificationList errors; 
-					if (!sm->checkPwdValidation(value.toString(), &errors, LanguageManager::getInstance()->getFreeCatalog(LANG_EN))) {
-						jsonErrorsArray.add("User.password isn't valid");
-						jsonErrorsArray.add(errors.getErrorsArray());
+					std::string old_password;
+					auto old_password_obj = updates->get("User.password_old");
+					if (old_password_obj.isEmpty()) {
+						jsonErrorsArray.add("User.password_old not found");
+					}
+					else if (!old_password_obj.isString()) {
+						jsonErrorsArray.add("User.password_old isn't a string");
 					}
 					else {
-						auto result_new_password = user->setNewPassword(value.toString());
-						
-						switch (result_new_password) {
-							// 0 = new and current passwords are the same
-						case 0: jsonErrorsArray.add("new password is the same as old password"); break;
-							// 1 = password changed, private key re-encrypted and saved into db
-						//case 1: extractet_values++; break;
-							// 2 = password changed, only hash stored in db, couldn't load private key for re-encryption
-						case 2: jsonErrorsArray.add("password changed, couldn't load private key for re-encryption");  break;
-							// -1 = stored pubkey and private key didn't match
-						case -1: jsonErrorsArray.add("stored pubkey and private key didn't match"); break;
+						old_password_obj.convert(old_password);
+					}
+
+					bool old_password_valid = false;
+					NotificationList errors;
+					if (old_password.size()) 
```

Comment:
`old_password` is always an empty string since it was never set?!

Reply:
three lines above it was set
```cpp
old_password_obj.convert(old_password);
````
old_password_obj is a json-object which has a method convert(T& varname)  which convert the content of the JSON-Object into the type of the given variable and fill it with that. 
The JSON lib Poco use isn't really intuitive I admit. 
Maybe I should use another one.